base/error.jl

```diff
@@ -49,3 +49,34 @@ macro assert(ex, msgs...)
     end
     :($(esc(ex)) ? $(nothing) : throw(Main.Base.AssertionError($msg)))
 end
+
+
+macro throw_argument_error(msg)
```

Comment:
This should not be a macro. It's much better to be a function. (and you don't want this to be inlined.)